web/src/redux/actions/articles-page/index.ts

```diff
@@ -16,212 +14,165 @@ import { urlLanguageRegEx } from "src/utils/language";
-        type: "UPDATE_ARTICLES",
-        payload: { list: [{ ...freshCurrentArticle, githubAuthors }] },
-      });
-    } catch (error) {
-      const freshCurrentArticle =
-        isLoaded(getState().articlesPage.currentArticle) || loadedCurrentArticle;
-      dispatch({
-        type: "UPDATE_ARTICLES_PAGE",
-        payload: { currentArticle: { ...freshCurrentArticle, githubAuthors: "ERROR" } },
-      });
-      Sentry.captureException(error, { tags: { type: "WEB_FETCH" } });
-    }
-  };
+export const fetchCurrentArticleAuthors = async (): Promise<void> => {
+  const { currentArticle } = getState().articlesPage;
+  const loadedCurrentArticle = isLoaded(currentArticle);
+
+  // Don't re-fetch data again
+  if (!loadedCurrentArticle || isLoaded(loadedCurrentArticle.githubAuthors)) return;
+
+  try {
+    actions.articlesPage.set({ currentArticle: { ...loadedCurrentArticle, githubAuthors: null } });
+
+    const githubAuthors = (
+      await Promise.all(
+        loadedCurrentArticle.authors?.map((author) => {
+          return fetchV2("api:GithubUsers/:login", {
+            params: { login: author },
+          });
+        }) || [],
+      )
+    ).map((response) => {
+      return response.user;
+    });
+    //  getting the current article from a fresh state
+    const freshCurrentArticle =
+      isLoaded(getState().articlesPage.currentArticle) || loadedCurrentArticle;
+    // update our page state
+    actions.articlesPage.set({ currentArticle: { ...freshCurrentArticle, githubAuthors } });
+    // update our cache state
+    actions.articles.set({ list: [{ ...freshCurrentArticle, githubAuthors }] });
+  } catch (error) {
+    const freshCurrentArticle =
+      isLoaded(getState().articlesPage.currentArticle) || loadedCurrentArticle;
+    actions.articlesPage.set({
+      currentArticle: { ...freshCurrentArticle, githubAuthors: "ERROR" },
+    });
+    Sentry.captureException(error, { tags: { type: "WEB_FETCH" } });
+  }
+};
 
 /**
  * Fetches the content of the current article
  */
-export const fetchCurrentArticle =
-  (): ThunkResult<ArticlesPageState | ArticlesState> => async (dispatch, getState) => {
-    const match = matchPath<{ lang?: LanguageEntity["code"]; slug: string }>(
-      history.location.pathname,
-      { path: `${urlLanguageRegEx}/Articles/:slug(.*)` },
-    );
-    const slug = match?.params.slug.replace(/\/$/, "") || "";
+export const fetchCurrentArticle = async (): Promise<void> => {
+  const match = matchPath<{ lang?: LanguageEntity["code"]; slug: string }>(
+    history.location.pathname,
+    { path: `${urlLanguageRegEx}/Articles/:slug(.*)` },
+  );
+  const slug = match?.params.slug.replace(/\/$/, "") || "";
+
+  const cashedArticle = hasInCollection<Article>(getState().articles.list, "slug", slug, [
+    ["content"],
+  ]);
+  if (cashedArticle) {
+    // update our page state
+    actions.articlesPage.set({ currentArticle: cashedArticle });
+
+    // Fetch authors
+    fetchCurrentArticleAuthors();
```

Comment:
- Could we bring back the dispatch for readability?

Reply:
I think it's better this way to improve DX

Reply:
> Could we bring back the dispatch for readability?

if the reason is **readability**, I suggest we tackle that differently than by just redundantly calling `dispatch`, maybe we can make change the function name to make it obvious that it's an Async Action? eg:
```suggestion
    thunkFetchCurrentArticleAuthors();
```

I can make these changes if you agree, or if you have more feedback let me know, but I don't think I will bring back the `dispatch`.

Reply:
introducing redundancy for readability isn't a good solution. let's resolve in later PRs